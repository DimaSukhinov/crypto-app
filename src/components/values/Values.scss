.values {

  &__header {
    display: flex;
    justify-content: space-between;
    padding-bottom: 10px;
    border-bottom: 1px solid;
  }

  &__rank, &__changes, &__add-header {
    min-width: 65px;
  }

  &__name, &__price, &__symbol {
    min-width: 130px;
  }

  @media (max-width: 630px) {
    &__name, &__changes {
      display: none;
    }
  }
}
