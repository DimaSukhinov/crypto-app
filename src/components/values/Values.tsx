import React, {useCallback} from 'react';
import './Values.scss';
import {ValuesType} from '../../App';

type ValueListPropsType = {
    values: ValuesType[]
    navigateToValue: (id: string) => void
}

export const Values = React.memo((props: ValueListPropsType) => {

    const openValuePage = useCallback((id: string) => () => props.navigateToValue(id), [props])

    return (
        <div className="valueList">
            {
                props.values.map(v => <div className={'valueList__value'} onClick={openValuePage(v.id)}>
                    <div className={'valueList__rank'}>
                        {v.rank}
                    </div>
                    <div className={'valueList__symbol'}>
                        {v.symbol}
                    </div>
                    <div className={'valueList__name'}>
                        {v.name}
                    </div>
                    <div className={'valueList__price'}>
                        {+(+v.priceUsd).toFixed(2)} $
                    </div>
                    <div className={'valueList__changePercent'}>
                        {+(+v.changePercent24Hr).toFixed(2)}%
                    </div>
                    <div className={'valueList__add'}>
                        Add
                    </div>
                </div>)
            }
        </div>
    );
})
