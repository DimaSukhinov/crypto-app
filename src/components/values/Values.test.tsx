import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Values } from './Values';

test('values components render', async () => {
  render(
    <Values values={[]} openAddModal={jest.fn()} navigateToValue={jest.fn()} />,
  );

  expect(screen.getByTestId('values-list')).toBeInTheDocument();
  expect(screen.getByTestId('pagination')).toBeInTheDocument();
  expect(screen.getByTestId('values-page')).toMatchSnapshot();
});
