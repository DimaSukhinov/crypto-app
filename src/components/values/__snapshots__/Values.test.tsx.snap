// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Values values components render 1`] = `
<div
  data-testid="values-page"
>
  <div
    class="values-list"
    data-testid="values-list"
  />
  <div
    class="pages"
    data-testid="pagination"
  />
</div>
`;

exports[`values components render 1`] = `
<div
  data-testid="values-page"
>
  <div
    class="values-list"
    data-testid="values-list"
  />
  <div
    class="pages"
    data-testid="pagination"
  />
</div>
`;
